cypress-io/cypress
# chore(webkit): add before:browser:launch and download support

- For #6422 

### User facing changelog

n/a

### Additional details

- Adds `before:browser:launch` support. `launchOptions.extensions` isn't supported since WK doesn't take extensions.
- Adds support for `config.downloadsFolder` and download events.
- Also fixes the issue where Cypress would reopen in interactive mode after `cypress run` ended, since it was making local system-tests fail.
- [x] Depends on #23579 

### Steps to test

Refer to #23579 testing steps for instructions on installing WebKit.

1. Run a project with `before:browser:launch` using WebKit, observe that the event is hit and that the launchOptions can be edited.
2. Run a project that downloads files, note that they are saved to `downloadsFolder` when using WebKit.
3. Exit a Cypress process after launching WebKit, note that Cypress no longer re-opens itself back up in `open` mode.

### How has the user experience changed?


### PR Tasks


- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
